crates/services/shared-sequencer/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+
+        let mut minimum_gas_price = ss_config.minimum_gas_price;
+
+        if let Some(index) = minimum_gas_price.find('.') {
+            minimum_gas_price.truncate(index);
+        }
+        let gas_price: u128 = minimum_gas_price.parse()?;
+        // Ceil the gas price to the next integer.
+        let gas_price = gas_price.saturating_add(1);
+
+        Ok(Self {
+            topic,
+            endpoints,
+            account_prefix,
+            coin_denom,
+            chain_id,
+            gas_price,
+        })
+    }
+
+    /// Returns the Cosmos account ID of the sender.
+    pub fn sender_account_id<S: Signer>(&self, signer: &S) -> anyhow::Result<AccountId> {
+        let sender_public_key = signer.public_key();
+        let sender_account_id = sender_public_key
+            .account_id(&self.account_prefix)
+            .map_err(|err| anyhow!("{err:?}"))?;
+
+        Ok(sender_account_id)
+    }
+
+    fn tendermint(&self) -> anyhow::Result<tendermint_rpc::HttpClient> {
+        Ok(tendermint_rpc::HttpClient::new(
+            &*self.endpoints.tendermint_rpc_api,
+        )?)
+    }
+
+    /// Retrieve latest block height
+    pub async fn latest_block_height(&self) -> anyhow::Result<u32> {
+        Ok(self
+            .tendermint()?
+            .abci_info()
+            .await?
+            .last_block_height
+            .value()
+            .try_into()?)
+    }
+
+    /// Retrieve account metadata by its ID
+    pub async fn get_account_meta<S: Signer>(
+        &self,
+        signer: &S,
+    ) -> anyhow::Result<AccountMetadata> {
+        let sender_account_id = self.sender_account_id(signer)?;
+        http_api::get_account(&self.endpoints.blockchain_rest_api, sender_account_id)
+            .await
+    }
+
+    /// Retrieve the topic info, if it exists
+    pub async fn get_topic(&self) -> anyhow::Result<Option<TopicInfo>> {
+        http_api::get_topic(&self.endpoints.blockchain_rest_api, self.topic).await
+    }
+
+    /// Post a sealed block to the sequencer chain using some
+    /// reasonable defaults and the config.
+    /// This is a convenience wrapper for `send_raw`.
+    pub async fn send<S: Signer>(
+        &self,
+        signer: &S,
+        account: AccountMetadata,
+        order: u64,
+        blob: Vec<u8>,
+    ) -> anyhow::Result<()> {
+        let latest_height = self.latest_block_height().await?;
+
+        self.send_raw(
+            latest_height.saturating_add(100),
+            signer,
+            account,
+            order,
+            self.topic,
+            Bytes::from(blob),
+        )
+        .await
+    }
+
+    /// Post a blob of raw data to the sequencer chain
+    #[allow(clippy::too_many_arguments)]
+    pub async fn send_raw<S: Signer>(
+        &self,
+        timeout_height: u32,
+        signer: &S,
+        account: AccountMetadata,
+        order: u64,
+        topic: [u8; 32],
+        data: Bytes,
+    ) -> anyhow::Result<()> {
+        let dummy_amount = Coin {
```

Comment:
Can we get a comment here about the reason of this dummy computation first before the real one ?